connect/workflows/core.py

```diff
@@ -280,47 +286,43 @@ async def run(self) -> Optional[dict]:
             logger.trace(f"Running {self.__class__.__name__}")
             await self.transform()
             await self.persist()
-            results = await self.transmit()
+            result = await self.transmit()
             await self.synchronize()
-            return {"lfh_message": self.message, "transmit_result": results}
+            return self._set_response(result)
         except Exception as ex:
+            logger.trace(f"Exception during run(): {str(ex)}")
             msg = await self.error(ex)
             raise Exception(msg)
 
-    def set_response(self, response: Response, result: dict):
+    def _set_response(self, transmit_result: dict) -> Union[Response, Dict]:
         """
-        Set the HTTP response based on the result of handling the message.  The result will be of the form:
-            {
-                "lfh_message": LinuxForHealthDataRecordResponse,
-                "transmit_result": List[dict]
-            }
+        Return the result of handling the message.  The result from calling _set_response will be either
+        self.message (Dict) or a Response instance containing the body and status code from the transmit results,
+        along with the LinuxForHealth-MessageId message header.
 
-        where lfh_message is the resulting LinuxForHealth message and transmit_result is an array of 0 or more transmit
-        result dicts, of the form:
+        transmit_result is an array of 0 or more transmit result dicts, of the form:
```

Comment:
if the docstring is correct, do we need to adjust the type hint for transmit result to `List[Dict]` ?

Reply:
Also, re: retransmit, I think this method is essentially the _retransmit_message() you suggest:

async def do_retransmit(message: dict, queue_pos: int):

The trick is that this method is called with a new incoming retransmit message or from the retransmit task, but it could be _do_retransmit (internal only).